src/lib/components/button/main-button.svelte

```diff
@@ -0,0 +1,91 @@
+<script lang="ts">
+  const getButtonState = (disabled?: boolean) => {
+    if (disabled) {
+      return 'disabled';
+    }
+    return 'default';
+  };
+
+  const getButtonStyle = (variant) => {
+    switch (variant) {
+      case 'secondary':
+        return 'secondary';
+      case 'destroy':
+        return 'destroy';
+      case 'primary':
+      default:
+        return 'primary';
+    }
+  };
+
+  const getButtonSize = (size) => {
+    switch (size) {
+      case 'small':
+        return 'small';
+      case 'default':
+      default:
+        return 'default';
+    }
+  };
+
+  export let disabled: boolean = false;
+  export let variant: string = '';
```

Comment:
Do we actually _need_ `getButtonStyle` or can we have TypeScript just handle this here?

```ts
export let variant: 'primary' | 'secondary | 'destroy' = 'secondary';
```

This will enforce the correct property and likely made it so that you can completely get rid of `getButtonStyle`.